tests/coverage/branch.coverage

```diff
@@ -0,0 +1,120 @@
+   LL|       |#![feature(coverage_attribute)]
+   LL|       |// edition: 2021
+   LL|       |// compile-flags: -Zunstable-options -Cinstrument-coverage=branch
+   LL|       |
+   LL|       |macro_rules! do_not_merge_with_signature {
+   LL|       |    () => { if true {} else {} };
+  ------------------
+  |  Branch (LL:16): [True: 15, False: 0]
+  ------------------
+  |  Branch (LL:16): [True: 15, False: 0]
+  ------------------
+  |  Branch (LL:16): [True: 15, False: 0]
+  ------------------
+  |  Branch (LL:16): [True: 15, False: 0]
+  ------------------
+   LL|       |}
+   LL|       |
+   LL|     15|fn if_not(cond: bool) {
+   LL|     15|    do_not_merge_with_signature!();
+   LL|       |
+   LL|       |    if
+   LL|       |        !
+   LL|     15|        cond
+  ------------------
+  |  Branch (LL:9): [True: 10, False: 5]
+  ------------------
+   LL|      5|    {
+   LL|      5|        println!("cond was false");
+   LL|     10|    } else {
+   LL|     10|        println!("cond was true");
+   LL|     10|    }
+   LL|     15|}
+   LL|       |
+   LL|     15|fn if_and(a: bool, b: bool) {
+   LL|     15|    do_not_merge_with_signature!();
+   LL|       |
+   LL|       |    if
+   LL|     15|        a
+  ------------------
+  |  Branch (LL:9): [True: 12, False: 3]
+  ------------------
+   LL|       |        &&
+   LL|     12|        b
+  ------------------
+  |  Branch (LL:9): [True: 8, False: 4]
+  ------------------
+   LL|      8|    {
+   LL|      8|        println!("both");
+   LL|      8|    } else {
+   LL|      7|        println!("not both");
+   LL|      7|    }
+   LL|     15|}
+   LL|       |
+   LL|     15|fn if_or(a: bool, b: bool) {
+   LL|     15|    do_not_merge_with_signature!();
+   LL|       |
+   LL|       |    if
+   LL|     15|        a
+  ------------------
+  |  Branch (LL:9): [True: 12, False: 3]
+  ------------------
+   LL|       |        ||
+   LL|      3|        b
+  ------------------
+  |  Branch (LL:9): [True: 14, False: 1]
+  ------------------
```

Comment:
Here the line count for `b` is correct, but the branch count's *true* arm is blatantly wrong, since it should be 2 instead of 14.

Reply:
Ah, I think what's happened here is that the count is (12 + 2 = 14).

The marker for `b == true` gets injected into the *then* block that both `a == true` and `b == true` branch to, so it has the execution count for both paths, which is not what we want.

I need to have the RHS inject its *true* marker into a separate block, which then jumps to the *then* block.